integrationtest/showcase/library/src/main/java/org/hibernate/search/integrationtest/showcase/library/repository/impl/DocumentRepositoryImpl.java

```diff
@@ -144,4 +147,25 @@ public long count() {
 
 		return query.getResultList();
 	}
+
+	@Override
+	public List<String> getAuthorsOfBooksHavingTerms(String terms, SortOrder order) {
+		FullTextSearchTarget<Document> target = entityManager.search( Document.class );
+		FullTextQuery<List<?>> query = target.query()
+				.asProjections(
+						target.projection().field( "author", String.class ).toProjection()
+				)
+				.predicate( f -> f.bool( b -> {
+					b.must( f.match()
+							.onField( "title" ).boostedTo( 2.0f )
+							.orField( "summary" )
+							.matching( terms )
+					);
+				} ) )
```

Comment:
You don't need a boolean query here, since you're not combining anything. This would work just fine:

```
				.predicate( f -> f.match()
					.onField( "title" ).boostedTo( 2.0f )
					.orField( "summary" )
					.matching( terms )
				) 
```